### PR TITLE
Add CI pipeline with Github Actions

### PR DESCRIPTION
**Pull request by GiulioRomualdi**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -29,7 +29,14 @@ jobs:
...
       run: |
-        vcpkg.exe install --triplet x64-windows gtest eigen3
+        vcpkg.exe install --triplet x64-windows gtest
+        hg clone https://bitbucket.org/eigen/eigen
+        cd eigen
+        md build
+        cd build
+        cmake ..
+        cmake --build . --config Release --target INSTALL
+        SET Eigen3_DIR=C:/Program Files (x86)/Eigen/lib/cmake/eigen3/
```
Comment (diegoferigo) [inline .github/workflows/ci.yml]: nitpick: if you pass shell: bash the shell is Git Bash and you can use familiar commands also on windows

**Comment by S-Dafarra** [comment]
I guess we do need this :grin:

**Comment by GiulioRomualdi** [comment]
On windows is still failing with the following error 
```
2020-05-14T10:42:58.3914037Z D:\a\osqp-eigen\osqp-eigen\tests\MPCTest.cpp(75,14): error C2065: 'M_PI': undeclared identifier [D:\a\osqp-eigen\osqp-eigen\build\tests\MPCUnitTests.vcxproj]
```
even if I added 
https://github.com/robotology/osqp-eigen/blob/bb6361f5326da52aa0fc972f46432405537bbc6c/tests/MPCTest.cpp#L20-L21

Do you have any hints? @S-Dafarra @traversaro

**Comment by traversaro** [comment]
> 
> 
> On windows is still failing with the following error
> 
> ```
> 2020-05-14T10:42:58.3914037Z D:\a\osqp-eigen\osqp-eigen\tests\MPCTest.cpp(75,14): error C2065: 'M_PI': undeclared identifier [D:\a\osqp-eigen\osqp-eigen\build\tests\MPCUnitTests.vcxproj]
> ```
> 
> even if I added
> 
> https://github.com/robotology/osqp-eigen/blob/bb6361f5326da52aa0fc972f46432405537bbc6c/tests/MPCTest.cpp#L20-L21
> 
> Do you have any hints? @S-Dafarra @traversaro

Either the  `math.h` or `cmath` header has been transitevly included by the headers included before that  line in that compilation unit. To fix this, either move the  definition of `_USE_MATH_DEFINES` before any other include, or pass it via command line (i.e. via CMake's `target_compile_definitions`).

**Pull request merged**